File: src/main/java/com/codelog/lastlife/events/DeathHandler.java
package com.codelog.lastlife.events;

import com.codelog.lastlife.LifeContext;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.TextColor;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDeathEvent;

public class DeathHandler implements Listener {

    @EventHandler
    public void onEntityDeathEvent(EntityDeathEvent event) {
        if (event.getEntity() instanceof Player p) {
            int remainingLives = LifeContext.get().getPlayerLives(p.getUniqueId()) - 1;
            if (remainingLives == 1) {
                Bukkit.getServer().dispatchCommand(Bukkit.getServer().getConsoleSender(),
                        "team join red " + p.getName());
            }

            if (remainingLives <= 0) {
                Bukkit.getServer().dispatchCommand(Bukkit.getServer().getConsoleSender(),
                        "team leave " + p.getName());
                p.sendMessage(ChatColor.RED + "You have lost all of your lives!");
                LifeContext.get().removePlayer(p.getUniqueId());
                return;
            }

            LifeContext.get().setPlayerLives(p.getUniqueId(), remainingLives);
            ChatColor color = (remainingLives == 1) ? ChatColor.RED : ChatColor.GREEN;
            p.sendMessage(ChatColor.YELLOW + "You have " + color + remainingLives + ChatColor.YELLOW + " lives left!");

            if (event.getEntity().getKiller() == null)
                return;

            if (event.getEntity().getKiller().getUniqueId() == LifeContext.get().getBogeyman()) {
                Player b = Bukkit.getServer().getPlayer(LifeContext.get().getBogeyman());
                if (b == null)
                    return;

                LifeContext.get().setBogeyman(null);

                b.sendMessage(
                        Component.text("You are no longer the bogeyman.")
                        .color(TextColor.color(0x55ff55))
                );
            }
        }
    }

}

File: src/main/kotlin/com/codelog/lastlife/events/DeathHandler.kt
package com.codelog.lastlife.events

import com.codelog.lastlife.LastLife
import com.codelog.lastlife.LifeContext
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.TextColor
import org.bukkit.Bukkit
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityDeathEvent

class DeathHandler : Listener {
    @EventHandler
    fun onEntityDeathEvent(event: EntityDeathEvent) {
        if (event.entity is Player) {
            val remainingLives = LifeContext.getPlayerLives(event.entity.uniqueId) - 1
            if (remainingLives == 1)
                LastLife.redTeam?.addEntry(event.entity.name)

            if (remainingLives <= 0) {
                LastLife.redTeam?.removeEntry(event.entity.name)
                LastLife.greenTeam?.removeEntry(event.entity.name)

                event.entity.sendMessage(
                    Component.text("You have lost all of your lives!", TextColor.color(0xff0000))
                )
            }

            LifeContext.setPlayerLives(event.entity.uniqueId, remainingLives)

            val killer = event.entity.killer ?: return

            if (killer.uniqueId == LifeContext.bogeyman) {
                if (LifeContext.bogeyman == null)
                    return

                val bogeyman = Bukkit.getServer().getPlayer(LifeContext.bogeyman!!)
                bogeyman?.sendMessage(
                    Component.text("You are no longer the bogeyman.", TextColor.color(0x55ff55))
                )
            }
        }
    }
}